index.js

```diff
@@ -0,0 +1,62 @@
+var gSettingsBinding = require('./build/Release/nodegsettings.node');
+
+// Constructor
+function GSettings( schemaId ) {
+	if ( ! schemaId.match(/^([a-zA-z0-9\-_]+\.)+[a-zA-z0-9\-_]+$/) )
+		throw new Error("'"+schemaId+"' is not a valid schema id!");
+
+	if ( ! gSettingsBinding.schema_exists(schemaId) )
+		throw new Error("Shema '"+schemaId+"' is not installed!");
+
+	this.schemaId = schemaId;
+}
+
+// Instance methods
+GSettings.prototype.get = function( key ) {
+	if ( ! key.match(/^[a-z0-9\-]+$/) )
+		throw new Error("Invalid key name format '"+key+"' !");
+
+	return gSettingsBinding.get_gsetting( this.schemaId, key );
+};
+
+GSettings.prototype.set = function( key, value ) {
+	if ( ! key.match(/^[a-z0-9\-]+$/) )
+		throw new Error("Invalid key name format '"+key+"' !");
+
+	if ( typeof value === 'undefined' )
+		throw new Error("Value is undefined!");
```

Comment:
Useful behaviour here would be to punch through to g_settings_reset here - of course, this behaviour should be documented carefully in the API!